### DISCUSSION TITLE
Unexpected behavior with the Horizontal Scrollbar for DataTable

### DISCUSSION THREAD

**Discussion by morelop** [Textualize/textual#4281] [Q&A] [answered]
Hi,

I have a DataTable, which is too wide to fit my terminal. So I get the Horizontal Scrollbar, which is as I expect.

But I do see some unexpected behavior.
1. Even if the my terminal has enough lines to show both the table and the horizontal scrollbar, no extra line is used for the scroll bar, so it hides the last line in the table. Here I would expect that if there are lines enough, the scrollbar would not hide the last line.
2. If I put the DataTable in a Horizontal container, and include a Footer, then Horizontal scrollbar completely disappears if there are too few lines in the terminal. Here I would expect the scroll bar to remain, on the line above the Footer.

Below code demonstrate the behavior. I have tried this with Windows 10, using both the old "Command Prompt", as well as "Command Prompt" and "Windows PowerShell" in the "Windows Terminal". I am using Textual version 0.52.1.

Is this expected behavior?

Regards
Morten

```
from textual.app import App
from textual.widgets import Footer, DataTable
from textual.containers import Horizontal

class MyApp(App):
    def compose(self):
        if True: # Toggle here to switch between the two cases
            # In a small terminal, the horizontal scroll bar seems to be lost?
            # If window is enough lines, the horizontal scroll bar is there
            with Horizontal():
                yield DataTable()
            yield Footer()
        else:
            # Even if the terminal is more than 30 lines, the horizontal scrollbar consumes the last line in the table
            yield DataTable()

    def on_mount(self):
        table = self.query_one(DataTable)
        for col in range(0,20):
            table.add_column("col{0}".format(col))
        for row in range(0,30):
            data = ["{0}x{1}".format(row,c) for c in range(0,20)]
            table.add_row(*data)

app = MyApp()
app.run()
```

**Comment by TomJGooding** [comment, marked as answer]
> 1, Even if the my terminal has enough lines to show both the table and the horizontal scrollbar, no extra line is used for the scroll bar, so it hides the last line in the table. Here I would expect that if there are lines enough, the scrollbar would not hide the last line.

I think this is the expected behaviour - see https://github.com/Textualize/textual/issues/3556#issuecomment-1777122066.

> 2. If I put the DataTable in a Horizontal container, and include a Footer, then Horizontal scrollbar completely disappears if there are too few lines in the terminal. Here I would expect the scroll bar to remain, on the line above the Footer.

I think this has possibly revealed an issue with this FAQ - https://textual.textualize.io/FAQ/#why-doesnt-the-datatable-scroll-programmatically.

If you change the `max-height` of the DataTable to 100%, this seems to work as expected. It apparently was changed at some point to 100vh.

**Reply by morelop** [reply]
Thank you for the prompt reply.
That seems to solve my problem.

Regards
Morten
